third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -131,6 +128,13 @@ impl StructType {
 pub struct CachedStructIndex(pub usize);
 
 #[derive(Debug, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct StructName {
+    pub module: ModuleId,
+    pub name: Identifier,
+}
+
+#[derive(Derivative, Ord, PartialOrd)]
+#[derivative(Debug, Clone, Eq, Hash, PartialEq)]
```

Comment:
Should we merkle-ize this w. cryptohash? Wasn't this where we were also replacing nodes by copying the whole trees? Would forcing a type to always have corresponding hash (that can be used for equality) be useful? Alternatively, could the compression crate be useful?
@ziaptos @runtian-zhou @zekun000 @brmataptos

Reply:
My take is that compression crate will definitely be useful for storing the names here as names are structs that duplicates itself a lot. Compressing this type may make less sense because type instantiation can generate different arbitrary many types so would be hard if we don't have an efficient deallocation system.